tests/robustness/coordinate/path_lock_test.go

```diff
@@ -0,0 +1,151 @@
+package coordinate
+
+import (
+	"math/rand"
+	"path/filepath"
+	"sync"
+	"testing"
+	"time"
+)
+
+func TestPathLockBasic(t *testing.T) {
+	pl := NewPathLock()
+
+	for ti, tc := range []struct {
+		name  string
+		path1 string
+		path2 string
+	}{
+		{
+			name:  "same path",
+			path1: "/a/b/c",
+			path2: "/a/b/c",
+		},
+		{
+			name:  "path2 is a descendent of path1",
+			path1: "/a/b/c",
+			path2: "/a/b/c/d/e",
+		},
+		{
+			name:  "path1 is a descendent of path2",
+			path1: "/a/b/c/d/e",
+			path2: "/a/b/c",
+		},
```

Comment:
If the underlying implementation is based on `filepath` or any path based parser, then you should probably include all the cases in your doc. 
If it is string based then this if fine.

Reply:
Done, test cases now match the package documentation